src/library/playlistfeature.cpp

```diff
@@ -164,16 +186,38 @@ void PlaylistFeature::buildPlaylistList() {
 
     for (int row = 0; row < playlistTableModel.rowCount(); ++row) {
         int id = playlistTableModel.data(
-            playlistTableModel.index(row, idColumn)).toInt();
-        QString name = playlistTableModel.data(
-            playlistTableModel.index(row, nameColumn)).toString();
-        int count = playlistTableModel.data(
-            playlistTableModel.index(row, countColumn)).toInt();
-        int duration = playlistTableModel.data(
-            playlistTableModel.index(row, durationColumn)).toInt();
-        m_playlistList.append(qMakePair(id, QString("%1 (%2) %3")
-                                        .arg(name, QString::number(count),
-                                                mixxx::Duration::formatSeconds(duration))));
+                playlistTableModel.index(row, idColumn)).toInt();
+        m_playlistList.append(qMakePair(id, createPlaylistLable(
+                playlistTableModel, row, nameColumn, countColumn, durationColumn)));
+    }
+}
+
+void PlaylistFeature::reloadPlaylistInPlaylistList(int playlist_id) {
+    // Setup the sidebar playlist model
+    QSqlTableModel playlistTableModel(this, m_pTrackCollection->database());
+    playlistTableModel.setTable("PlaylistsCountsDurations");
+    QString filter = "id=" + QString::number(playlist_id);
+    playlistTableModel.setFilter(filter);
+    playlistTableModel.select();
+    while (playlistTableModel.canFetchMore()) {
+        playlistTableModel.fetchMore();
+    }
+    QSqlRecord record = playlistTableModel.record();
+    int nameColumn = record.indexOf("name");
+    int countColumn = record.indexOf("count");
+    int durationColumn = record.indexOf("durationSeconds");
+
+    DEBUG_ASSERT(playlistTableModel.rowCount() <= 1);
+    if (playlistTableModel.rowCount() > 0) {
+        for (auto it = m_playlistList.begin();
```

Comment:
This loop now appears multiple times. Can't we extract it in a function named replacePlaylistLabel(id, label)? That would improve readability.

Reply:
A better name for `m_playlistList` would be `m_playlistListLabels`.

I noticed this only after your added the function for formatting the label. Before it wasn't obvious and hidden behind a pile of code. Inspecting the actual type in the header is too far away.

Reply:
`buildPlaylistList` -> `createPlaylistLabels()`
`reloadPlaylistInPlaylistList` -> `updatePlaylistLabel(int playlistId)`
How much effort would it be to change the naming?

Reply:
Eclipse has a rename feature, so it should be no deal.